baremetal_network_provisioning/db/bm_nw_provision_db.py

```diff
@@ -376,42 +379,45 @@ def get_all_bnp_phys_switches(context):
     return switches
 
 
-def update_bnp_phys_switch_status(context, switch_id, switch):
+def update_bnp_phys_switch_status(context, switch_id, sw_status):
     """Update physical switch status."""
     try:
         with context.session.begin(subtransactions=True):
             (context.session.query(models.BNPPhysicalSwitch).filter_by(
                 id=switch_id).update(
-                    {'status': switch['status']},
+                    {'status': sw_status},
                     synchronize_session=False))
     except exc.NoResultFound:
         LOG.error('no physical switch found for id: %s', switch_id)
 
 
-def update_bnp_phys_switch_snmpv2(context, switch_id, switch):
-    """Update physical switch with snmpv2 params."""
+def update_bnp_phys_swport_status(context, swid, port_name, port_status):
+    """Update physical switch port status by switch id."""
     try:
         with context.session.begin(subtransactions=True):
-            (context.session.query(models.BNPPhysicalSwitch).filter_by(
-                id=switch_id).update(
-                    {'write_community': switch['write_community']},
+            (context.session.query(models.BNPPhysicalSwitchPort).filter_by(
+                switch_id=swid, interface_name=port_name).update(
+                    {'status': port_status},
                     synchronize_session=False))
     except exc.NoResultFound:
-        LOG.error('no physical switch found for id: %s', switch_id)
+        LOG.error('no phy switch port found for swid: %s port_name',
```

Comment:
All LOG.error , debug should be followed as per the latest neutron code. 
It should be i8n'ed.

here %s is missing for representing one more variable.
It is better to test the exception cases .